front/lib/api/files/tool_output.ts

```diff
@@ -68,3 +72,97 @@ export async function internalCreateToolOutputFile(
 
   return file;
 }
+
+export async function internalCreateSearchableTextFile(
+  auth: Authenticator,
+  {
+    title,
+    conversationId,
+    rows,
+  }: {
+    title: string;
+    conversationId: string;
+    rows: Array<CSVRecord>;
+  }
+): Promise<FileResource> {
+  const workspace = auth.getNonNullableWorkspace();
+  const user = auth.user();
+
+  let content: string = "";
+  const sections: Array<CoreAPIDataSourceDocumentSection> = [];
+
+  for (const row of rows) {
+    const rowId = typeof row["Id"] === "string" ? row["Id"] : "";
+    const rowName = typeof row["Name"] === "string" ? row["Name"] : "";
+    const rowContent = JSON.stringify(row);
+    const section: CoreAPIDataSourceDocumentSection = {
+      prefix: `${title} - Row ${rowId} ${rowName}`,
```

Comment:
Title should be a top-level prefix. The algorithm will take care of repeating it